testng-core/src/test/java/test/support/SafeRandoms.java

```diff
@@ -0,0 +1,23 @@
+package test.support;
+
+import java.security.SecureRandom;
+import java.util.Random;
+
+public final class SafeRandoms {
+
+  private static final Random random = new SecureRandom();
+
+  private SafeRandoms() {}
+
+  public static int nextInt(int bound) {
```

Comment:
Why do you use util method instead of an object?

Reply:
I didnt quite understand the concern here. Can you please elaborate?

Reply:
```
SafeRandom random = new SafeRandom();
int value = random.nextInt();
```

Reply:
Can you let me know what is the issue with the current implementation? You are suggesting inheritance and I am making use of composition to do the same thing. I am not sure if understand the benefits of having to use inheritance here. 

I need a random number which is NOT zero. Why wouldnt a utility method suffice here?

Reply:
Utils are imperative programming, not composition.

Composition here will be to create your own random that create another random.
Inheritance here will be to create your own random that extends another random.
Both approach are oop and valid.

Reply:
Inheritance is not going to allow me to use another flavor of random (if there's a need for it). The current implementation works with its own random implementation if that's what is needed but it also helps with allowing a user to inject the random as a dependency. I was hinting at composition in that aspect as in `I have a randomness object with me` which I believe is what composition stands for (`has a` vs `is a` relationship).

My point here is that, we don't need an OOP approach here but just a plain vanilla utility that helps in generating random numbers that aren't zero.

Reply:
That's the point. Everything is possible with utils methods. But why is jdk providing an object for random instead of few static methods?